tests/js/server/aql/aql-queries-simple.js

```diff
@@ -1335,6 +1335,18 @@ function ahuacatlQuerySimpleTestSuite () {
       });
     },
 
+    testLargeQuery : function() {
+      let q = "";
+      const cnt = 1000;
+      for (let i = 0; i < cnt; ++i) {
+        q += `LET v${i} = NOOPT(1)\n`;
+      }
+      q += "RETURN v0";
+      for (let i = 1; i < cnt; ++i) {
+        q += ` + v${1}`;
+      }
+      assertEqual([cnt], AQL_EXECUTE(q, {}, {optimizer: {rules: ['-all']}}).json);
+    }
   };
```

Comment:
I would like to ask for a third Test, pretty much again the same as you have.
However this time on a One-Shard Database (EE) with a Single Collection Access included.
The reasoning behind this is:
If we are on OneShard the query will be optimized onto a single DBServer, so the Splitting needs to take place on a different Server, not the Coordinator.

I would love to have such a test on a "normal" collection access, but it will be utterly hard to write an AQL containing 1000 nodes that are pushed into the same DBServer snippet and not be moved/optimized away. You cannot get away with all rules disabled here.

Reply:
Actually this should be relatively simple:
```
let q = "FOR d IN docs LET x = (\n";
const cnt = 900;
for (let i = 0; i < cnt; ++i) {
  q += `LET v${i} = NOOPT(1)\n`;
}
q += "RETURN d.v + v0";
for (let i = 1; i < cnt; ++i) {
  q += ` + v${i}`;
}
q += ")[0]\nSORT x RETURN x";
```
Assuming we have a collection `docs` containing documents with an attribute `v` ranging from 0..x this should return `[cnt..cnt + x]`. In a one-shard setup the whole query should be moved to the DB servers, and since all calculations use `NOOPT` nothing should be optimized away. I will add an according EE test.

Reply:
I agree on the one-shard setup.
I was looking for a non-one-shard test, but that will be hard to write to get a single snippet, will check the enterprise test, thanks for adding.